### PR TITLE
Addition of AMQP URI configuration

### PR DESCRIPTION
**Pull request by BigPeeg**
Updated to use the RabbitMQ hostname provided by the configuration.

### CODE REVIEW

**Review by MattCavey — APPROVED** [review]

**Comment by gargath** [comment]
I'm not too sharp (hur-hur) on my C#. Can you add a comment on how this configuration is supposed to be passed into the process when it starts?
Is this going to be an environment variable or command-line parameter that needs to be set?

File: Source/Service/Messaging/TransactionEventSender.cs [hunk 1]
```diff
@@ -16,13 +15,20 @@ public class TransactionEventSender : ITransactionEventSender, IDisposable
...
         private readonly IConnection _connection;
 
-        public TransactionEventSender()
+        public TransactionEventSender(IFileProcessorConfig fileProcessorConfig)
         {
-            var connectionFactory = new ConnectionFactory() { HostName = HostName };
+            if (fileProcessorConfig == null) throw new ArgumentNullException(nameof(fileProcessorConfig));
+            var connectionFactory = new ConnectionFactory()
+            {
+                HostName = fileProcessorConfig.ArchiveAdaptationRequestQueueHostname,
```
Comment (Diggers89) [inline Source/Service/Messaging/TransactionEventSender.cs:23]: Should be TransactionEventQueueHostName

File: Source/Service/Messaging/TransactionEventSender.cs [hunk 2]
```diff
@@ -16,13 +15,20 @@ public class TransactionEventSender : ITransactionEventSender, IDisposable
...
+                Port = fileProcessorConfig.ArchiveAdaptationRequestQueuePort,
```
Comment (Diggers89) [inline Source/Service/Messaging/TransactionEventSender.cs:24]: Should be TransactionEventQueuePort

File: Source/Service/Messaging/TransactionEventSender.cs [hunk 3]
```diff
@@ -16,13 +15,20 @@ public class TransactionEventSender : ITransactionEventSender, IDisposable
...
+                UserName = fileProcessorConfig.MessageBrokerUser,
+                Password = fileProcessorConfig.MessageBrokerPassword
+            };
             _connection = connectionFactory.CreateConnection();
             _channel = _connection.CreateModel();
 
-            Console.WriteLine($"Connection established to {HostName}");
+            Console.WriteLine($"TransactionEventSender Connection established to {fileProcessorConfig.ArchiveAdaptationRequestQueueHostname}");
```
Comment (Diggers89) [inline Source/Service/Messaging/TransactionEventSender.cs:31]: Same here

**Pull request merged**